src/adapter/src/coord/dataflows.rs

```diff
@@ -658,6 +658,7 @@ fn eval_unmaterializable_func(
         }
         UnmaterializableFunc::CurrentUser => pack(Datum::from(&*session.user().name)),
         UnmaterializableFunc::MzEnvironmentId => pack(Datum::from(&*state.config().environment_id)),
+        UnmaterializableFunc::MzCluster => pack(Datum::from(session.vars().cluster())),
```

Comment:
We should figure out how this PR would work with https://github.com/MaterializeInc/materialize/issues/15195 before merging. Should this new `mz_cluster()` function always return the session setting, or should it return the cluster on which the current statement is executing? If the latter, then we would need a way to communicate that from the peek plan to here. I think it could go into `ExprPrepStyle::OneShot` which seems fine?

Reply:
> Should this new mz_cluster() function always return the session setting, or should it return the cluster on which the current statement is executing?

The way I want to take #15195 is to introduce a new "statement scope" for settings, which defaults to the session scope but can be overridden with the `SETTING` clause. Then there's a very natural answer here: it should return the setting in the statement scope!